common/src/main/java/com/revenuecat/purchases/common/ReceiptInfo.kt

```diff
@@ -9,9 +9,8 @@ class ReceiptInfo(
     val purchaseOptionId: String? = null,
     val storeProduct: StoreProduct? = null,
 
-    // TODO BC5 - replace price/currency with making a single PricingPhase for amazon observermode products?
-    val price: Double? = null, // only passed for amazon observer mode
-    val currency: String? = null // only passed for amazon observer mode
+    val price: Double? = storeProduct?.oneTimeProductPrice?.priceAmountMicros?.div(MICROS_MULTIPLIER.toDouble()),
+    val currency: String? = storeProduct?.oneTimeProductPrice?.currencyCode
```

Comment:
might be good to add a test to confirm the logic in these...i think right now we just test that the receipt info passed in == the receipt info posted, which contains these values. but i'm thinking more something that confirms the currency is in fact the onetimeproductprice's currency

Reply:
Oh I'm rethinking this, when syncing observer mode amazon purchases, these could be the price/currency of a subscription I guess... But then, we probably should try to map that info into the `PricingPhases` class? Though that would not be possible without more info 🤔. I need to think more about this...

Reply:
@tonidero Why would Amazon observer mode subscriptions fall under the `oneTimeProductPrice` property? 🤔 I don't think I follow that 😅

Reply:
i _think_ i might be following. when we sync amazon observer mode purchases, we don't have a way of getting price/currency ourselves, so we require the user to pass it in. that's what these fields used to be for.

for any non-amazon-observermode purchases, we would just pass the price/currency in the storeproduct.

now, our backend is repurposing the price/currency field to be used for OTP (right?) so now we have a need for those to work for amazon observer mode purchases (both subs and inapps), AND for all other inapp purchases.

does that make sense?

Reply:
so we could rename these to be the onetimeProductPrice/Currency, but we still need a field for amazon observer mode products (potentially subs)

Reply:
so we need to think through how amazon observer mode subs pricing info would get passed.

okay those comments were just rambles, that took me some time to figure out. maybe that helps you understand hearing it differently @joshdholtz ?

Reply:
So, as part of the SDK API, we have a method [`syncObserverModeAmazonPurchase`](https://github.com/RevenueCat/purchases-android/blob/main/purchases/src/main/kotlin/com/revenuecat/purchases/Purchases.kt#L266). That's used so devs can give us price data for their purchases, subscriptions or not. Currently, we just store the price and currency parameters from that method in the `ReceiptInfo` `price` and `currency` fields. 

If we separate the price info between `oneTimeProductPrice` and `subscriptionPricingPhases` depending on the type of purchase, we wouldn't know how to store the data given by devs in `syncObserverModeAmazonPurchase` unless we ask devs for it.

Not sure if there is a different way to find that info from the SDK in observer mode though... I'm not that familiar with the Amazon store.

Reply:
And yeah @beylmk's rambles are correct 😛 . Sorry I missed some steps on my reasoning there 😅

Reply:
so it feels like we should at least think about whether amazon pricing phases will work before finalizing this OTP functionality. but i think we always knew amazon might cause some changes, so i'm OK merging this PR for now...

Reply:
Okay, so... I added some new tests to `ReceiptInfo` for when `price` and `currency` are manually passed in and it seems like what I have in this PR should work for Amazon still 🤞 

[syncObserverModeAmazonPurchase](https://github.com/RevenueCat/purchases-android/blob/main/purchases/src/main/kotlin/com/revenuecat/purchases/Purchases.kt#L266) doesn't pass in a `storeProduct` but only `productIds`, `price`, and `currency`. 

So I _tthhiiinnnnkkk_ this should be okay for all the use cases now? 😇

Reply:
i'm good with it! i'll link this PR in our amazon tickets so we remember this conversation

Reply:
Yeah, I think functionally this works. My only concern is about these fields being confusing with `pricingPhases`. I don't think that's a new problem from this PR so don't want to block this just because of that but I would add a comment or create a ticket to address in the future.

Reply:
BTW: Yes, it looks like for amazon receipts we are persisting the price sent in and use it.

Reply:
so we'll just need to make sure that we either add a different field or that backend has some way of differentiating between amazon prices vs google OTPS^^

@greenietea FYI

Reply:
> some way of differentiating between amazon prices vs google OTPS

We should be able to use the platform header for this, right?

Reply:
yeah, i'd think so!